### PR TITLE
Add follow-schema layout for exec

### PR DESCRIPTION
**Pull request by kevinmbeaulieu**
Implements #1265. See the issue for more details, but to summarize:

When `ExecLayout` is set to `follow-schema`, output generated code to a directory instead of a single file. Each file in the output directory will correspond to a single `*.graphql` schema file (plus a `root!.generated.go` file containing top-level definitions that are not specific to a single schema file).

`ExecLayout` defaults to `single-file`, which is the current behavior, so this new functionality is opt-in.

These layouts expose similar functionality to the `ResolverLayout`s with the same name, just applied to `exec` instead of `resolver`.

I have:
 - [x] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [x] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

**Comment by kevinmbeaulieu** [comment]
Status update: still working on adding more tests to this and fixing some bugs with some edge cases

**Comment by kevinmbeaulieu** [comment]
Status update: I started adding more unit tests for this change, but found a couple corner cases where my current implementation is failing. I've had to put this on hold for now to focus on other work, but I'll try to get back to it when I do have more time. Otherwise, someone else is welcome to pick it up where I've left off too! :)

**Comment by kevinmbeaulieu** [comment]
@vektah I finally got back to this and got everything working as far as I can tell. It looks like a big change so I broke it up into a few commits to make it easier to review:
1. Add Visual Studio Code settings files to .gitignore (alongside IntelliJ IDEA)
2. Create a separate `ExecConfig` type based on `PackageConfig` but with additional fields to support the new layouts
3. Implement two layouts: `single-file`, which corresponds to the baseline functionality of outputting all code to a single `generated.go` file, and `follow-schema`, which implements the functionality described in #1265.
4. Add unit tests to verify that the code generated by `single-file` and `follow-schema` is functionally equivalent (i.e. defines all the same functions, types, etc., just organized differently). This was the main challenge that took months to figure out since there weren't any existing tools that quite served this need, so I developed a new tool [eq-go](https://github.com/kevinmbeaulieu/eq-go/) to handle comparing the packages generated by the two layouts. As part of this change, I basically created two copies of the code in `codegen/testserver`, in `codegen/testserver/singlefile` and `codegen/testserver/followschema` (and then deleted the original copy). So each of these directories tests the corresponding layout, and then there's a `codegen/testserver/generated_test.go` that runs eq-go to ensure the two directories are equivalent.

There seems to be one CI check still failing that I'm not sure how to fix, so any advice there would be appreciated.

**Comment by kevinmbeaulieu** [comment]
@MichaelJCompton @lwc I'd been talking to @vektah last year about this change, but haven't heard from him so not sure if he's still active with gqlgen, so tagging you since you both look like you've been merging changes recently. Would love if one of you could take a look (or tag someone if there's someone specific that should review it). See #1265 for more context on the proposed change.

**Comment by asif-mahmud** [comment]
any plan to merge this pr ?

**Comment by kevinmbeaulieu** [comment]
@asif-mahmud I would love to, if one of the maintainers can review it! 😄

**Comment by kevinmbeaulieu** [comment]
Seeing a few remaining test failures that look like they are related to my changes- will take a look tomorrow

**Comment by kevinmbeaulieu** [comment]
@StevenACoffman -  Just updated the tests and they all pass locally now, if you want to trigger a final build and then merge. Thanks for your help getting this over the finish line!

**Comment by StevenACoffman** [comment]
There were some test failures, so I rebased and regenerated so they passed. I hope that's ok. Thanks!

File: codegen/generate.go [hunk 1]
```diff
@@ -14,3 +39,176 @@ func GenerateCode(data *Data) error {
...
+		SubscriptionRoot: data.SubscriptionRoot,
+		AllDirectives:    data.AllDirectives,
+	}
+}
+
+// Root file contains top-level definitions that should not be duplicated across the generated
+// files for each schema file.
+func generateRootFile(data *Data) error {
+	dir := data.Config.Exec.DirName
+	path := filepath.Join(dir, "root!.generated.go")
```
Comment (rthille) [inline codegen/generate.go:130]: We're seeing issues with our tooling due to the '!' in this filename. Any reason why it has to have an exclamation point in it?  Quoting with "" doesn't even help, so it can be tricky to deal with.
```
bash$ echo "!root"
bash: !root: event not found
bash$ echo '!root'
!root
bash$
```

Comment (kevinmbeaulieu) [reply codegen/generate.go:130]: My original intention was to minimize potential naming conflicts in case people had schema files named `root.graphql` in their project, but no reason it needs to be an exclamation point- could be changed to another symbol or just accept the possibility of conflicts

Comment (yar00001) [reply codegen/generate.go:130]: Thank you @kevinmbeaulieu - just sent a PR to change the name to `root_.generated.go`
https://github.com/99designs/gqlgen/pull/1701
